### PR TITLE
BC prop - vol menu revamp

### PR DESCRIPTION
**Pull request by NoSloppy**
Replace cycle-through with Quick MAX and MIN actions.
Add rotate to increase / decrease volume.
Limit blade angle to avoid erratic rotational volume changes at extreme blade angles.
Change sound_library prompts to use to good ol' PlayPolyphonic to allow for faster playback/overlap when quick adjustments are made.

### CODE REVIEW

File: props/saber_BC_buttons.h [hunk 1]
```diff
@@ -442,91 +457,122 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
...
       if (SFX_voldown) {
-        sound_library_.SayVolumeDown();
+        hybrid_font.PlayPolyphonic(&SFX_voldown);
       } else {
         beeper.Beep(0.10, 2000);
         beeper.Beep(0.20, 1500);
       }
       STDOUT.print("Volume Down - Current Volume: ");
       STDOUT.println(dynamic_mixer.get_volume());
+    } else QuickMinVolume();
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:517]: Please change to:

} else {
  QuickMinVolume();
}

File: props/saber_BC_buttons.h [hunk 2]
```diff
@@ -661,7 +707,7 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         } else {
           On();
         }
-      }
+      } else QuickMaxVolume();
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:710]: also fix like above

File: props/saber_BC_buttons.h [hunk 3]
```diff
@@ -795,7 +818,7 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         STDOUT<< "Battery Voltage: " << battery_monitor.battery() << "\n";
         STDOUT<< "Battery Percentage: " <<battery_monitor.battery_percent() << "\n";
         SaberBase::DoEffect(EFFECT_BATTERY_LEVEL, 0);
-      }
+      } else QuickMinVolume();
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:821]: and here

Comment (NoSloppy) [reply props/saber_BC_buttons.h]: did them all.

File: props/saber_BC_buttons.h [hunk 4]
```diff
@@ -442,91 +457,126 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
...
+    dynamic_mixer.get_volume() - VOLUME * 0.90));
+    if (SFX_volmin) {
+      hybrid_font.PlayPolyphonic(&SFX_volmin);
+      } else {
+        beeper.Beep(0.5, 1000);
+      }
+      STDOUT.print("Minimum Volume \n");
+  }
+
+  void DetectMenuTurn() {
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:547]: It seems wasteful to have a separate function for detecting volume changes, isn't this mostly the same as the other menu detection code? Why have two?

File: props/saber_BC_buttons.h [hunk 5]
```diff
@@ -2,8 +2,8 @@
 saber_BC_buttons.h
   http://fredrik.hubbe.net/lightsaber/proffieos.html
   Copyright (c) 2016-2019 Fredrik Hubinette
-  Copyright (c) 2021 Brian Conner with contributions by:
-  Fredrik Hubinette, Fernando da Rosa, and Matthew McGeary.
+  Copyright (c) 2023 Brian Conner with contributions by:
+  Fredrik Hubinette, Fernando da Rosa, and Matthew McGeary and caiwyn.
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:6]: and, and

File: props/saber_BC_buttons.h (hunk 4 above)
Comment (NoSloppy) [reply props/saber_BC_buttons.h:547]: Sorry, I don't follow. Same as what other code, VolumeMenu() ?
VolumeMenu() just handles enter/exit.
DetectMenuTurn() is all motion stuff.
Did I double up on a function?

File: props/saber_BC_buttons.h (hunk 5 above)
Comment (NoSloppy) [reply props/saber_BC_buttons.h]: edited.

**Pull request merged**